python/dolfinx/nls/petsc.py

```diff
@@ -23,9 +29,29 @@
 
 from dolfinx import cpp as _cpp
 from dolfinx import fem
-from dolfinx.fem.petsc import create_matrix, create_vector
+from dolfinx.fem.forms import form as _create_form
+from dolfinx.fem.petsc import (
+    apply_lifting,
+    assemble_matrix_block,
+    assemble_matrix_nest,
+    assemble_vector,
+    assemble_vector_block,
+    copy_block_vec_to_functions,
```

Comment:
These functions names don't read well and there are a lot. In most cases (maybe all?) a single function would suffice for one direction of copying - the necessary operation can be inferred from the `Vec` type and whether a `Function` or a list of `Function`s  is passed. 


A simple way to reduce the number would be to pass an argument.

Reply:
Again, how do we differentiate between block and nest copy? There is no clear way of distinguishing them on the PETSc level. If you have one way of doing it, please show it. 

This also then goes back to you wanting to remove the "Function input in `copy_function_to_vec`, which if you make it a PETSc vector, there is no way of distinguishing the copy direction.

Reply:
The PETSc Mat/Vec has a `getType()` method, which can be used to check if it is blocked. 

The direction can be inferred by the argument order.

Reply:
Nope, see Jack's PR: https://github.com/FEniCS/dolfinx/pull/3652
And you couldn't infer the direction if one would go for your suggestion above:
https://github.com/FEniCS/dolfinx/pull/3648#discussion_r1973473558
> No need to pass a dolfinx.fem.Function here, only PETSc vec from it is needed.
as the operation would be PETSc.Vec <-> PETSc.Vec. 
However, lets just review Jack's PR instead of discussing here.